sumatra/recordstore/shelve_store.py

```diff
@@ -38,7 +38,13 @@ class ShelveRecordStore(RecordStore):
 
     def __init__(self, shelf_name=".smt/records"):
         self._shelf_name = shelf_name
+        # Some shelve backends add an extension to the filename, and more than one
```

Comment:
The problem was uncovered by adding the `recordstore.clear()` test, so in that sense it is covered at least indirectly. A full test would be painful to implement given the variety of shelve backends on different systems, and is low priority given than few people use `ShelveRecordStore`.